lib/matplotlib/backend_bases.py

```diff
@@ -1595,6 +1595,73 @@ def __init__(self, figure):
         self.toolbar = None  # NavigationToolbar2 will set me
         self._is_idle_drawing = False
 
+    def _repr_html_(self):
+        if not self.figure.axes and not self.figure.lines:
+            return
+
+        dpi = self.figure.dpi
+        is_retina = False
+
+        import IPython
+        ip = IPython.get_ipython()
```

Comment:
This would fail whenever IPython is not available.

Reply:
Wouldn't `_repr_html_` only get triggered for ipython?

Reply:
I guess IPython invented this; but in principle anything can try to call such method if it's available.

Reply:
The other way you can check this is `if 'IPython' in sys.modules`.  That way if the user has not previously imported IPython we won't do it for them.